client/src/app/services/barman-service-number/barman-service-number.component.ts

```diff
@@ -0,0 +1,35 @@
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { Barman } from '../../_models';
+import { MatPaginator, MatSort, MatTableDataSource } from '@angular/material';
+import { BarmanService, ServiceService } from '../../_services';
+
+@Component({
+    selector: 'app-barman-service-number',
+    templateUrl: './barman-service-number.component.html',
+})
+export class BarmanServiceNumberComponent implements OnInit {
+
+    displayedColumns = ['nickname', 'number'];
+    barmenData: MatTableDataSource<Barman>;
+
+    @ViewChild(MatSort) sort: MatSort;
+    @ViewChild(MatPaginator) paginator: MatPaginator;
+
+    constructor(private barmanService: BarmanService, private serviceService: ServiceService) { }
+
+    ngOnInit(): void {
+        this.serviceService.getWeek().subscribe(week => {
+            this.barmanService.getAll().subscribe(barmen => {
+                barmen.filter(barman => barman.active).map(barman => {
+                    this.barmanService.getServices(barman.id, week.start, week.end).subscribe(services => {
+                        barman.services = services;
+                    });
+                    return barman;
+                });
+                this.barmenData = new MatTableDataSource(barmen);
+                this.barmenData.paginator = this.paginator;
+                this.barmenData.sort = this.sort;
+            });
+        });
```

Comment:
We will need to review our use of RxJS, it clearly not good this way (having a function inside another, inside another,...). We should look into RxJS to find a way! Otherwise, we could just transform everything into promise (And add `toPromise()` after http call).

Anyway :smile:, I will probably create an issue for this, no more callback hell is needed now :wink:

Reply:
You are right. I don't really know how to improve our RxJS usage.

Another problem: several API calls from different component are the same...